airflow/models/baseoperator.py

```diff
@@ -240,72 +240,123 @@ def partial(
+
+    # Override None kwargs which don't have a dag default value by Airflow default value
+    partial_kwargs["owner"] = partial_kwargs["owner"] or DEFAULT_OWNER
+    partial_kwargs["trigger_rule"] = partial_kwargs["trigger_rule"] or DEFAULT_TRIGGER_RULE
+    partial_kwargs["depends_on_past"] = (
+        partial_kwargs["depends_on_past"] if partial_kwargs["depends_on_past"] is not None else False
+    )
+    partial_kwargs["ignore_first_depends_on_past"] = (
+        partial_kwargs["ignore_first_depends_on_past"]
+        if partial_kwargs["ignore_first_depends_on_past"] is not None
+        else DEFAULT_IGNORE_FIRST_DEPENDS_ON_PAST
+    )
+    partial_kwargs["wait_for_past_depends_before_skipping"] = (
+        partial_kwargs["wait_for_past_depends_before_skipping"]
+        if partial_kwargs["wait_for_past_depends_before_skipping"] is not None
+        else DEFAULT_WAIT_FOR_PAST_DEPENDS_BEFORE_SKIPPING
+    )
+    partial_kwargs["wait_for_downstream"] = (
+        partial_kwargs["wait_for_downstream"] if partial_kwargs["wait_for_downstream"] is not None else False
+    )
+    partial_kwargs["retries"] = (
+        partial_kwargs["retries"] if partial_kwargs["retries"] is not None else DEFAULT_RETRIES
+    )
+    partial_kwargs["queue"] = partial_kwargs["queue"] or DEFAULT_QUEUE
+    partial_kwargs["pool_slots"] = (
+        partial_kwargs["pool_slots"] if partial_kwargs["pool_slots"] is not None else DEFAULT_POOL_SLOTS
+    )
+    partial_kwargs["execution_timeout"] = (
+        partial_kwargs["execution_timeout"] or DEFAULT_TASK_EXECUTION_TIMEOUT
+    )
+    partial_kwargs["retry_delay"] = (
+        partial_kwargs["retry_delay"] if partial_kwargs["retry_delay"] is not None else DEFAULT_RETRY_DELAY
+    )
+    partial_kwargs["retry_exponential_backoff"] = (
+        partial_kwargs["retry_exponential_backoff"] if partial_kwargs["retry_exponential_backoff"] else False
+    )
+    partial_kwargs["priority_weight"] = (
+        partial_kwargs["priority_weight"] if partial_kwargs["priority_weight"] else DEFAULT_PRIORITY_WEIGHT
+    )
+    partial_kwargs["weight_rule"] = partial_kwargs["weight_rule"] or DEFAULT_WEIGHT_RULE
+    partial_kwargs["inlets"] = partial_kwargs["inlets"] or []
+    partial_kwargs["outlets"] = partial_kwargs["outlets"] or []
 
     # Post-process arguments. Should be kept in sync with _TaskDecorator.expand().
     if "task_concurrency" in kwargs:  # Reject deprecated option.
         raise TypeError("unexpected argument: task_concurrency")
     if partial_kwargs["wait_for_downstream"]:
         partial_kwargs["depends_on_past"] = True
-    partial_kwargs["start_date"] = timezone.convert_to_utc(partial_kwargs["start_date"])
-    partial_kwargs["end_date"] = timezone.convert_to_utc(partial_kwargs["end_date"])
+    partial_kwargs["start_date"] = timezone.convert_to_utc(partial_kwargs["start_date"])  # type: ignore
+    partial_kwargs["end_date"] = timezone.convert_to_utc(partial_kwargs["end_date"])  # type: ignore
```

Comment:
Why are these ignores needed?

Reply:
Since I didn't defined the type of `partial_kwargs`, python auto detected it to Dict[str, <combination of all the values types>] which is not compatible with the used functions, so I fixed it by defining the type as Dict[str, Any]